talismane_core/src/main/java/com/joliciel/talismane/AnnotationObserver.java

```diff
@@ -0,0 +1,39 @@
+///////////////////////////////////////////////////////////////////////////////
+//Copyright (C) 2016 Joliciel Informatique
+//
+//This file is part of Talismane.
+//
+//Talismane is free software: you can redistribute it and/or modify
+//it under the terms of the GNU Affero General Public License as published by
+//the Free Software Foundation, either version 3 of the License, or
+//(at your option) any later version.
+//
+//Talismane is distributed in the hope that it will be useful,
+//but WITHOUT ANY WARRANTY; without even the implied warranty of
+//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//GNU Affero General Public License for more details.
+//
+//You should have received a copy of the GNU Affero General Public License
+//along with Talismane.  If not, see <http://www.gnu.org/licenses/>.
+//////////////////////////////////////////////////////////////////////////////
+package com.joliciel.talismane;
+
+import java.util.List;
+
+/**
+ * For getting notified whenever annotations are added.
+ * 
+ * @author Assaf Urieli
+ *
+ */
+public interface AnnotationObserver {
+	/**
+	 * An event fired whenever annotations are added to the observed subject.
+	 * 
+	 * @param subject
+	 *            the text to which annotations are about to be added
+	 * @param annotations
+	 *            the annotations getting added
+	 */
+	public <T> void beforeAddAnnotations(AnnotatedText subject, List<Annotation<T>> annotations);
```

Comment:
If we have a hook after addition, it allows for observers not to perform the addition itself if it needs the result, even though it is predictable. It avoids duplicating this computation. I’d say that such a hook wouldn’t need the newly added annotations but only get the resulting annotated text.